### PR TITLE
Limit the amount of memory that a chunked HTTP/1.1 request will allocate

### PR DESCRIPTION
**Pull request by bryancall**

### CODE REVIEW

File: proxy/http/HttpTunnel.cc [hunk 1]
```diff
@@ -1151,6 +1151,10 @@ HttpTunnel::producer_handler(int event, HttpTunnelProducer *p)
 
   // Handle chunking/dechunking/chunked-passthrough if necessary.
   if (p->do_chunking) {
+    if (p->chunked_handler.chunked_buffer && p->chunked_handler.chunked_buffer->max_read_avail() > 524288) {
```
Comment (maskit) [inline proxy/http/HttpTunnel.cc]: Why didn't you define a constant for this? It's the same issue with the H2 issue, right?

Comment (bryancall) [reply proxy/http/HttpTunnel.cc]: Yes, it is a similar issue.  I can do that.

Comment (zwoop) [reply proxy/http/HttpTunnel.cc]: Yeh, +1 on at least having one central define for this (instead of a configuration option).

**Comment by SolidWallOfCode** [comment]
I think it might be better to make the existing flow control mechanism work, rather than having any hardwired constant. As far as I can tell, that mechanism does check the chunked buffer for memory usage, or could be easily extended to do so.

**Comment by bryancall** [comment]
This PR hasn't been updated in more then 2 months and I am closing out PRs that haven't had any updates.  Please feel free to reopen if want to update it.

**Pull request closed without merging**